software/base/src/main/java/brooklyn/entity/software/MachineLifecycleEffectorTasks.java

```diff
@@ -550,6 +575,27 @@ public void stop() {
      * If no errors were encountered call {@link #postStopCustom()} at the end.
      */
     public void stop(ConfigBag parameters) {
+        doStop(parameters, new Callable<StopMachineDetails<Integer>>() {
```

Comment:
Now that `MachineLifecycleEffectorTasks` is configurable it could end up being persisted. Don't introduce new anonymous classes even if no references to them are kept, so that existing anonymous classes keep their indexes.
From what I see the anonymous classes are task bodies, so they are not supposed to get persisted, but better be safe than sorry.